src/sql/serialize.rs

```diff
@@ -0,0 +1,953 @@
+//! Database serialization module.
+//!
+//! The module contains functions to serialize database into a stream.
+//!
+//! Output format is based on [bencoding](http://bittorrent.org/beps/bep_0003.html).
+
+/// Database version supported by the current serialization code.
+///
+/// Serialization code MUST be updated before increasing this number.
+///
+/// If this version is below the actual database version,
+/// serialization code is outdated.
+/// If this version is above the actual database version,
+/// migrations have to be run first to update the database.
+const SERIALIZE_DBVERSION: &str = "99";
```

Comment:
So, we'll have to update serializing and deserializing everytime we change the db scheme, and (de)serializing only works between the same core version. We should make sure that everyone is OKish with this trade-off before merging.